homeassistant/components/screenlogic/number.py

```diff
@@ -0,0 +1,86 @@
+"""Support for a ScreenLogic number entity."""
+import logging
+
+from screenlogicpy.const import BODY_TYPE, DATA as SL_DATA, EQUIPMENT, SCG
+
+from homeassistant.components.number import NumberEntity
+
+from . import ScreenlogicEntity
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORTED_SCG_NUMBERS = (
+    "scg_level1",
+    "scg_level2",
+)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up entry."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    equipment_flags = coordinator.data[SL_DATA.KEY_CONFIG]["equipment_flags"]
+    if equipment_flags & EQUIPMENT.FLAG_CHLORINATOR:
+        async_add_entities(
+            [
+                ScreenLogicNumber(coordinator, scg_level)
+                for scg_level in coordinator.data[SL_DATA.KEY_SCG]
+                if scg_level in SUPPORTED_SCG_NUMBERS
+            ]
+        )
+
+
+class ScreenLogicNumber(ScreenlogicEntity, NumberEntity):
+    """Class to represent a ScreenLogic Number."""
+
+    def __init__(self, coordinator, data_key, enabled=True):
+        """Initialize of the entity."""
+        super().__init__(coordinator, data_key, enabled)
+        self._body_type = SUPPORTED_SCG_NUMBERS.index(self._data_key)
+        self._attr_max_value = SCG.LIMIT_FOR_BODY[self._body_type]
+
+    @property
+    def name(self) -> str:
+        """Name of the number."""
+        return f"{self.gateway_name} {self.sensor['name']}"
```

Comment:
```suggestion
        self._attr_name = f"{self.gateway_name} {self.sensor['name']}"
```

https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L299